src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,111 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+
+    private Node<K, V>[] table;
+    private int capacity;
+    private int size;
+
+    public MyHashMap() {
+        this.capacity = DEFAULT_CAPACITY;
+        this.table = (Node<K, V>[]) new Node[capacity];
+        this.size = 0;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        boolean keyPresent = updateIfKeyPresent(key, value);
+        if (!keyPresent) {
+            if (size >= capacity * DEFAULT_LOAD_FACTOR) {
+                resize();
+            }
+            createNewNode(key, value);
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        Node<K, V> node = getNodeKey(key);
+        return node == null ? null : node.value;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int hash(Object key) {
+        if (key == null) {
+            return 0;
+        }
+        int hash = key.hashCode() % capacity;
+        return hash < 0 ? hash + capacity : hash;
```

Comment:
I could be wrong, but what if adding capacity won't be enough? Why not use Math.abs()?